src/engine/renderer/tr_shader.cpp

```diff
@@ -3571,7 +3578,51 @@ static bool ParseShader( const char *_text )
 
 			continue;
 		}
-		// parallax mapping
+		// normalFormat <format>
+		//
+		// expects OpenGL format as default
+		//
+		// OpenGL format is described there:
+		//     https://github.com/KhronosGroup/glTF/tree/2.0/specification/2.0#materialnormaltexture
+		//
+		// > The normal vectors use OpenGL conventions where +X is right and +Y is up.
+		// > +Z points toward the viewer.
+		//
+		// example of formats:
+		//
+		//  1  1  1  OpenGL format (default)
+		//  1 -1  1  DirectX format with reverted green (Y) channel
+		// -1  1  1  weird other format with reverted red (X) channel
+		else if ( !Q_stricmp( token, "normalFormat" ) )
+		{
+			for ( int i = 0; i < 3; i++ )
+			{
+				token = COM_ParseExt2( text, false );
+
+				if ( token[ 0 ] == 0 )
+				{
+					Log::Warn("missing normalFormat parm in shader '%s'", shader.name );
+					continue;
+				}
+
+				if ( !Q_stricmp( token, "-1" ) )
+				{
+					shader.normalFormat[ i ] = -1;
+				}
+				else if ( !Q_stricmp( token, "1" ) )
+				{
+					// do nothing, that's the default
+				}
+				else
+				{
+					Log::Warn("unknown normalFormat keyword '%s' not supported in '%s'", token, shader.name );
+					SkipRestOfLine( text );
+					continue;
```

Comment:
I think you want `break` here.

Reply:
`break` would stop right there the parsing of the material, hence disabling it entirely. This would not be a warning but an error, and such syntax issue is not problematic enough to entirely invalidate the material (at least diffuse, glow and specular would be properly rendered and it may be rendered with a flipped reflection on y, with a warning).

There was a similar talk in #80 (about `return` and shader keyword not parm, but that's almost the same).

`SkipRestOfLine` is there to be sure next material line is read.

Reply:
No, `continue` or `break` applies to the innermost loop. In this case that's the `for ( int i = 0; i < 3; i++ )` for the normal directions, which are all supposed to be on one line, right? So if you have skipped the rest of the line then you should break out of the loop.

Reply:
Oh, you're right, I'm dumb. Fixed.